### ISSUE TITLE
Show an error message on an invalid named module

### ISSUE THREAD

**Issue by ghivert** [gleam-lang/gleam#3972]
Hi!

Currently, when the LSP opens a Gleam file whose name is incorrect (something like `my-module.gleam` for instance), nothing happens. It could be nice to let the LSP indicates it as a bug (because `gleam build` will fail) with a message like "my-module is not a valid name for a Gleam module. You should switch to my_module instead."

**Comment by lpil** [comment]
Good idea! Thank you

**Comment by ahuangg** [comment]
hello @lpil, would I be able to pick up this issue?

**Comment by lpil** [comment]
That would be wonderful! Yes please!

**Comment by ahuangg** [comment]
I was thinking of implementing the module name validation in the language server [engine.rs](https://github.com/gleam-lang/gleam/blob/main/compiler-core/src/language_server/engine.rs#L143C4-L162C6) file within the compile function. The idea is to check if module names are valid similar to [this](https://github.com/gleam-lang/gleam/blob/main/compiler-core/src/build/package_loader.rs#L205) and, if not, add them to the results.  Before moving forward and submitting a pull request, I'd love to hear your thoughts on this approach and if there is an alternative solution I missed.

**Comment by lpil** [comment]
Wouldn't we want an error to be shown for the command line too? I could see it being useful in that case also

**Comment by ImmemorConsultrixContrarie** [comment]
Is this actually an issue? Doesn't feel like something that's worth making language server more complicated.

Error when trying to import the module:
```
error: Syntax error
  ┌─ /wibble/src/wibble.gleam:3:11
  │
3 │ import bad-module
  │           ^ I was not expecting this

Found `-`, expected one of: 
- An import, const, type, or function.
```

Error when trying to name the package
```
error: File IO failure

An error occurred while trying to parse this file:

    /wibble/gleam.toml

The error message from the file IO library was:

    TOML parse error at line 1, column 8
  |
1 | name = "bad-module"
  |        ^^^^^^^^^^^^
Package names may only contain lowercase letters, numbers, and underscores
```

Error when trying to compile package named `bad_module` with file `/wibble/src/bad-module.gleam`:
```
warning: Invalid module name

Module names must begin with a lowercase letter and contain only lowercase
alphanumeric characters or underscores.

Hint: Rename `/wibble/src/bad-module.gleam` to
be
valid, or remove this file from the project source.

  Compiling bad_module
   Compiled in 0.49s
error: Module does not exist

Module `bad_module` was not found.

Hint: Try creating the file `src/bad_module.gleam`.
```

@lpil are there any other ways a badly named module can be used?


<details><summary>Questions in case it actually is an issue</summary>
<p>

1. Where exactly is language server supposed to report problem in the buffer? First line and first character? What if the file is empty?
2. While investigating, I've found this funny thing, where gleam happily compiles nothingness:
```
$ ls src/ # empty src/
$ gleam check
   Compiled in 0.47s
$ gleam build
   Compiled in 0.56s
$ gleam run
   Compiled in 0.53s
error: Module does not exist

Module `wibble` was not found.

Hint: Try creating the file `src/wibble.gleam`.
```
I guess that's a problem with having no difference between library and binary? Hence why everything compiles, but then `gleam run` cannot find the entry point for binary.
I dunno if this should be an issue either. It's such an edge case, and it's not like it breaks anything.
</p>
</details>

**Comment by lpil** [comment]
Yes, those are how we currently report invalid module names. This issue is for the scenario outlined in ghivert's opening message. We have nothing for that presently.
